nucliadb_telemetry/nucliadb_telemetry/batch_span.py

```diff
@@ -110,11 +110,15 @@ def on_end(self, span: ReadableSpan) -> None:
             if not self._spans_dropped:
                 logger.warning("Queue is full, likely spans will be dropped.")
                 self._spans_dropped = True
+            return
 
         try:
             self.queue.put_nowait(span)  # type: ignore
         except asyncio.QueueFull:
-            logger.warning(f"Queue is full. Queue size : {self.queue.qsize()}")
+            if not self._spans_dropped:
+                logger.warning(f"Queue is full. Queue size : {self.queue.qsize()}")
+                self._spans_dropped = True
+            return
```

Comment:
should there be a `else` branch here where we set `self._spans_dropped = False`?

Reply:
no no, we want that to be cleared later on when the batch is done

this is to prevent too many logging calls